app/controllers/login_controller.rb

```diff
@@ -74,26 +74,13 @@ def refresh
     end
 
   def revoke
-    # # Populate the request body with the defined parameters from the docs
-    # # https://developers.procore.com/reference/authentication#revoke-token
-    # request = {
-    #   client_id: ENV['CLIENT_ID'],
-    #   client_secret: ENV['CLIENT_SECRET'],
-    #   token: session[:oauth_response]['access_token']
-    # }
-    # # Send a request to revoke the access token to Procore and return to the Sign In page.
-    # # Note that in the production environment, this request uses the BASE_URL (https://api.procore.com) to revoke the token.
-    # response = RestClient.post(ENV['BASE_URL'] + '/oauth/revoke', request.to_json)
-    # # Deletes current session to clear out session variables
-    # reset_session
-    # # Redirects user back to the login page
-    # redirect_to login_index_path
-
-    # rescue RestClient::ExceptionWithResponse
-    #   if session[:oauth_response]
-    #     redirect_to users_home_path, danger: "Something went wrong. Please refresh your access token and try again."
-    #   else
-    #     redirect_to login_index_path, danger: "Something went wrong. Please try again."
-    #   end
+    # Send a request to revoke the access token to Procore
+    client.revoke
+    # Deletes current session to clear out session variables
+    reset_session
+    # Redirects user back to the login page
+    redirect_to login_index_path, success: "Token was successfully revoked"
+  rescue Procore::Error
+    redirect_to login_index_path, danger: "Something went wrong. Please try again."
```

Comment:
Could this cause an infinite loop?

Let me explain: the `login_index` is displayed to users who are not logged in.

But in this case, if `client.revoke` raises a `Procore::Error`, then the `reset_sessions` line is skipped.
Therefore the user might still have a session, which would try to redirect the user _out_ of the `login_index` page and possibly cause an infinite redirection.

Reply:
That's a good thought, but it isn't a problem because having a session does not cause the user to be redirected out of `login_index`. They could log in twice, regardless of whether they still have a session.